connection/connection.go

```diff
@@ -10,4 +13,9 @@ type Connection interface {
 	GetOfflineQueue() *[]types.QueryObject
 	GetState() *int
 	EmitEvent(int, interface{})
+	RegisterRoom(string, string, types.IRoom)
+	UnregisterRoom(string)
+	GetRequestHistory() *map[string]time.Time
+	RenewSubscriptions()
+	GetRooms() map[string]map[string]types.IRoom
```

Comment:
This `map[string]map[string]types.IRoom` is used 8 times in this PR, wouldn't a `types.RoomList` be nice to declare in types?